src/services/search/service/search-index.ts

```diff
@@ -231,3 +231,110 @@ export const ingest = function (
     payload: GIngest;
   }
 >;
+
+/**
+ * Search Index Role document
+ *
+ * @see https://docs.globus.org/api/search/reference/role_list/#role
+ */
+export type IndexRole = {
+  id: string;
+  creation_date: string;
+  role_name: string;
+  index_id: string;
+  principal: string;
+  principal_type: string;
+};
+
+/**
+ * Search Index Role List document
+ *
+ * @see https://docs.globus.org/api/search/reference/role_list/#rolelist
+ */
+export type IndexRoleListResponse = {
+  role_list: IndexRole[];
+};
+
+/**
+ * Get role list for a search index as owner or admin only
+ *
+ * @see https://docs.globus.org/api/search/reference/role_list/
+ */
+export const getRoles = function (
```

Comment:
To match the other services and resources, I think this should actually be `search.index.roles(.getAll | .create | .remove)`

https://globus.github.io/globus-sdk-javascript/modules/_globus_sdk.html#service-method-names

Reply:
Ah oops, I'll get that fixed up

Reply:
Okay, I think it's okay now. I'm not totally sure about the module export from `search/index.ts` though so please double check me there

Reply:
👍 - I just swapped the changes in `search/index.js` for: https://github.com/globus/globus-sdk-javascript/pull/425/files#diff-9b2324bba31437d1ca01432b814e10bcf00777d71d1ffc0a23d3b2481bda0132R10

Should have the same effect, but keep things structured.